### PR TITLE
Infinity abilities bug fix for buddy.js

### PR DESCRIPTION
**Pull request by SirPhoenix88**
Fixes the bug. Limits number of abilities but sets the number to more than players should ever need per stage part.

### CODE REVIEW

**Comment by ThauEx** [comment]
Does this new limit apply to normal attack too?

**Comment by SirPhoenix88** [comment]
After testing, it appears to apply to normal attack as well. I'm trying to figure out how to get it to apply to just the abilities.

**Comment by SirPhoenix88** [comment]
This is strange, as the original pre-infinity syntax did not do this, I just increased the amount.

**Comment by ThauEx** [comment]
Your first MR has doen the same.

**Comment by ThauEx** [comment]
You could do a check (untested)
```
if (json[buddy].ability_panels[panel].ability_id != '30151001') {
[...]
}
```

original JSON:
```
			"ability_panels": [{
				"panel_no": 1,
				"num": 0,
				"uid": "4872449569062912",
				"max_num": "0",
				"name": "Attack",
				"assets": {
					"command_icon_30151001": "/Content/lang/ab_image/ability/30151001/30151001_128.png"
				},
				"ability_ss_point": "50",
				"ability_id": "30151001"
			},
```

File: lib/filter/buddy.js [hunk 1]
```diff
@@ -137,10 +138,13 @@ exports.update = function (json) {
...
      * Fills soulbreak gauge one full unit per ability
+     * Number of uses set to 20, setting to '0' (infinity) creates bugs, and they refill every round of a stage
      */
     for (var panel in json[buddy].ability_panels) {
-      json[buddy].ability_panels[panel].num = '0';
-      json[buddy].ability_panels[panel].max_num = '0';
+      if (json[buddy].ability_panels[panel].ability_id != '30151001') {
+        json[buddy].ability_panels[panel].num = '20';
+        json[buddy].ability_panels[panel].max_num = '20';
+      {
```
Comment (ThauEx) [inline lib/filter/buddy.js]: Wrong one, try `}`

**Comment by KHShadowrunner** [comment]
Interesting. I wonder what the bug is in regards to why it wouldn't do the same if it was over a number that the skill can possibly be. 

Either way, as originally planned, yep, I'd just do an if/else and avoid the first instance.

**Comment by ThauEx** [comment]
Can you please fix the closing curly brace?

**Comment by SirPhoenix88** [comment]
Sorry, had to step away from keyboard. Fixed!

**Comment by SirPhoenix88** [comment]
I think it is primarily because ability uses can grow as high as you have mythril/gems (Game over Phoenix bonus).

**Comment by SirPhoenix88** [comment]
I added a new feature that refills HP between rounds. It unintentionally also raises dead party members between rounds as well. (For those of us that don't use god stats.)

**Pull request merged**